consensus/src/payload_manager.rs

```diff
@@ -477,86 +488,125 @@ async fn get_transactions_for_observer(
     ))
 }
 
+async fn fetch_transactions<T: TDataInfo>(
+    batch_summary: &Vec<T>,
+    batch_reader: Arc<dyn BatchReader>,
+    block: &Block,
+    ordered_authors: &[PeerId],
+) -> ExecutorResult<Vec<SignedTransaction>> {
+    let mut vec_ret = Vec::new();
+    let batches_and_responders = batch_summary
+        .iter()
+        .map(|proof| {
+            let mut signers = proof.signers(ordered_authors);
+            if let Some(author) = block.author() {
+                signers.push(author);
+            }
+            (proof.info().clone(), signers)
+        })
+        .collect();
+    let timestamp = block.timestamp_usecs();
+    let receivers = QuorumStorePayloadManager::request_transactions(
+        batches_and_responders,
+        timestamp,
+        batch_reader,
+    );
+    for (digest, rx) in receivers {
+        match rx.await {
+            Err(e) => {
+                // We probably advanced epoch already.
+                warn!(
+                    "Oneshot channel to get a batch was dropped with error {:?}",
+                    e
+                );
+                return Err(DataNotFound(digest));
+            },
+            Ok(Ok(data)) => {
+                vec_ret.push(data);
+            },
+            Ok(Err(e)) => {
+                return Err(e);
+            },
+        }
+    }
+    let ret: Vec<SignedTransaction> = vec_ret.into_iter().flatten().collect();
+    Ok(ret)
+}
+
+async fn get_transactions(
+    batches_and_responders: Vec<(BatchInfo, Vec<PeerId>)>,
+    timestamp: u64,
+    batch_reader: Arc<dyn BatchReader>,
+) -> ExecutorResult<Vec<SignedTransaction>> {
+    let mut vec_ret = Vec::new();
+    let receivers = QuorumStorePayloadManager::request_transactions(
```

Comment:
it looks like a lot of duplicate code, why fetch_transaction not call get_transactions directly?

Reply:
I think `fetch_transaction` is not used. We only need `get_transactions`. I will clean it up.